crates/ide-completion/src/completions/dot.rs

```diff
@@ -1172,4 +1181,63 @@ impl<B: Bar, F: core::ops::Deref<Target = B>> Foo<F> {
         "#]],
         );
     }
+
+    #[test]
+    fn test_struct_function_field_completion() {
+        check(
+            r#"
+struct S { va_field: u32, fn_field: fn() }
+fn foo() { S { va_field: 0, fn_field: || {} }.fi$0() }
+"#,
+            expect![[r#"
+                fd fn_field fn()
+            "#]],
+        );
+
+        check_edit(
+            "fn_field",
+            r#"
+struct S { va_field: u32, fn_field: fn() }
+fn foo() { S { va_field: 0, fn_field: || {} }.fi$0() }
+"#,
+            r#"
+struct S { va_field: u32, fn_field: fn() }
+fn foo() { (S { va_field: 0, fn_field: || {} }).fn_field() }
```

Comment:
This is why I wanted the `check_edit` test :^) The output here is wrong, this should be
```suggestion
fn foo() { (S { va_field: 0, fn_field: || {} }.fn_field)() }
```
This is how you call fields in rust (to disambiguate from method calls)

Reply:
> Sorry for all the review rounds and thanks for sticking with it, this should be the last :)
No worries, this is just lesson for me to put good a PR next time.

I keep getting confused where to put the end parenthesis. I somehow finally decided this was correct thing, my bad.